File: msbversionupdater/src/main/java/jp/misyobun/lib/versionupdater/MSBUpdateInfo.java
package jp.misyobun.lib.versionupdater;

import java.io.Serializable;

/**
 * Created by usr0600259 on 15/02/05.
 */
public class MSBUpdateInfo  implements Serializable{

    /** 強制アップデート */
    public static final String TYPE_FORCE = "force";

    /** 前回の強制アップデートバージョン*/
    public String last_force_required_version;
    /** 最新バージョン */
    public String required_version;
    /** 対象URL */
    public String update_url;
    /** 強制:force 任意:optional */
    public String type;

    /**
     * コンストラクタ
     * @param last_force_required_version
     * @param requiredVersion
     * @param updateUrl
     * @param type
     */
    public MSBUpdateInfo(final String last_force_required_version,
                         final String requiredVersion,
                         final String updateUrl,
                         final String type) {
        this.required_version = requiredVersion;
        this.update_url = updateUrl;
        this.type = type;
        this.last_force_required_version = last_force_required_version;
    }
}

File: msbversionupdater/src/main/java/jp/misyobun/lib/versionupdater/MSBUpdateInfo.kt
package jp.misyobun.lib.versionupdater

import java.io.Serializable

/**
 *
 * Created by misyobun on 15/02/05.
 */
data class MSBUpdateInfo
/**
 * @param last_force_required_version
 * @param required_version
 * @param update_url
 * @param type
 */
(
        /** Last compulsory update version */
        var last_force_required_version: String,
        /** New Version  */
        var required_version: String,
        /** url  */
        var update_url: String,
        /** `force` or `optional`  */
        var type: String) : Serializable
